xdem/spstats.py

```diff
@@ -0,0 +1,702 @@
+    else:
+        return (psill1 + psill2) / std_err ** 2
+
+
+def integrate_fun(fun: Callable, low_b: float, upp_b: float) -> float:
+    """
+    Numerically integrate function between upper and lower bounds
+    :param fun: function
+    :param low_b: lower bound
+    :param upp_b: upper bound
+
+    :return: integral
+    """
+
+    return integrate.quad(fun,low_b,upp_b)[0]
+
+def cov(h: float,crange: float, model: str = 'Sph', psill: float = 1., kappa: float = 1/2, nugget: float = 0) -> Callable:
+    """
+    Covariance function based on variogram function (COV = STD - VGM)
+
+    :param h: spatial lag
+    :param crange: correlation range
+    :param model: model
+    :param psill: partial sill
+    :param kappa: smoothing parameter for Exp Class
+    :param nugget: nugget
+
+    :returns: covariance function
+    """
+
+    return (nugget + psill) - vgm(h,crange,model=model,psill=psill,kappa=kappa)
+
+def vgm(h: float, crange: float, model: str = 'Sph', psill: float = 1., kappa: float = 1/2, nugget: float = 0):
+    """
+    Compute variogram model function (Spherical, Exponential, Gaussian or Exponential Class)
+
+    :param h: spatial lag
+    :param crange: correlation range
+    :param model: model
+    :param psill: partial sill
+    :param kappa: smoothing parameter for Exp Class
+    :param nugget: nugget
+
+    :returns: variogram function
+    """
+
+    c0 = nugget #nugget
+    c1 = psill #partial sill
+    a1 = crange #correlation range
+    s = kappa #smoothness parameter for Matern class
+
+    if model == 'Sph':  # spherical model
+        if h < a1:
+            vgm = c0 + c1 * (3 / 2 * h / a1-1 / 2 * (h / a1) ** 3)
+        else:
+            vgm = c0 + c1
+    elif model == 'Exp':  # exponential model
+        vgm = c0 + c1 * (1-np.exp(-h / a1))
+    elif model == 'Gau':  # gaussian model
+        vgm = c0 + c1 * (1-np.exp(- (h / a1) ** 2))
+    elif model == 'Exc':  # stable exponential model
+        vgm = c0 + c1 * (1-np.exp(-(h/ a1)**s))
+
+    return vgm
+
+def std_err_finite(std: float, neff_tot: float, neff: float) -> float:
+    """
+    Standard error of subsample of a finite ensemble
+
+    :param std: standard deviation
+    :param neff_tot: maximum number of effective samples
+    :param neff: number of effective samples
+
+    :return: standard error
+    """
+    return std * np.sqrt(1 / neff_tot * (neff_tot - neff) / neff_tot)
+
+def std_err(std: float, neff: float) -> float:
+    """
+    Standard error
+
+    :param std: standard deviation
+    :param neff: number of effective samples
+
+    :return: standard error
+    """
+    return std * np.sqrt(1 / neff)
+
+def distance_latlon(tup1: tuple, tup2: tuple) -> float:
+    """
+    Distance between two lat/lon coordinates projected on a spheroid
+    :param tup1: lon/lat coordinates of first point
+    :param tup2: lon/lat coordinates of second point
+
+    :return: distance
+    """
+
+    # approximate radius of earth in km
+    R = 6373000
```

Comment:
Does this hardcoded number change the results significantly?

Either way, maybe it could be nice with a reference? A quick wikipedia-ing showed that it for example deviates from the WGS84 ellipsoid.

Reply:
This should at least be an argument with a default value.

Reply:
Added the reference + argument as default

Reply:
@rhugonnet I can't see that addition. Is it in a commit that is not yet pushed?